### PR TITLE
Add washer-controls cluster xml and generate zap

### PR DESCRIPTION
**Pull request by abeck-whirlpool**
Changes:
Add washer-controls cluster xml and generate zap.

Testing
Build all-clusters-app, chip-tool

Notes:
This contains changes for an in flight PR to address issues with Washer Controls which can be found here https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7029

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    <name>Washer Controls</name>
+    <code>0x0053</code>
+    <define>WASHER_CONTROLS_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>This cluster supports remotely monitoring and controling the different typs of functionality available to a washing device, such as a washing machine.</description>
+
+    <globalAttribute side="server" code="0xFFFD" value="1" />
+
+    <attribute side="server" code="0x0000" define="SPIN_SPEEDS"               type="ARRAY" entryType="STRING"                               writable="false" isNullable="false" optional="false">SpinSpeeds</attribute>
```
Comment (rbultman) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:43]:
```suggestion
    <attribute side="server" code="0x0000" define="SPIN_SPEEDS"               type="ARRAY" entryType="SpinSpeedIndex"                               writable="false" isNullable="false" optional="false">SpinSpeeds</attribute>
```

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    <cluster code="0x0053"/>
+    <field mask="0x01" name="Spin"/>
+    <field mask="0x02" name="Rinse"/>
+  </bitmap>
+  
+  <struct name="SpinSpeedIndex">
+    <cluster code="0x0053"/>
+    <item fieldId="0" name="Label" type="char_string" length="64" isNullable="false" writable="false"/>
+    <item fieldId="1" name="SpinSpeed" type="int8u" isNullable="false" writable="false"/>
+  </struct>
```
Comment (rbultman) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:30]: Reading ahead to the PR that removes SpinSpeedIndex, you don't need this anymore.
```suggestion
```

**Comment by bzbarsky-apple** [comment]
@rbultman @abeck-whirlpool please see https://github.com/project-chip/connectedhomeip/pull/26649

**Comment by abeck-whirlpool** [comment]
> > @rbultman @abeck-whirlpool please see #26649
> 
> Do we know why the PR was closed? It looks like it was closed by the PR author without comment.

This was closed as we were going to do the two PR's one for XML and codegen and another for the server addition (which will be handled by NXP) per the TCC recommendation

**Comment by abeck-whirlpool** [comment]
> @rbultman @abeck-whirlpool please see #26649

This was closed as we were going to do the two PR's one for XML and codegen and another for the server addition (which will be handled by NXP) per the TCC recommendation

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Review by rbultman — APPROVED** [review]

**Review by rbultman — APPROVED** [review]

**Comment by marktrayer** [comment]
There's something missing on the Java side, I'm not familiar enough with the tool chain to know what though (see the build failures).

**Review by marktrayer — APPROVED** [review]

File: src/controller/data_model/controller-clusters.matter [hunk 3]
```diff
@@ -2943,6 +2943,25 @@ client cluster RefrigeratorAndTemperatureControlledCabinetModeSelect = 82 {
...
 }
 
+/** This cluster supports remotely monitoring and controling the different typs of functionality available to a washing device, such as a washing machine. */
+client cluster WasherControls = 83 {
+  bitmap WasherControlFeature : BITMAP32 {
+    kSpin = 0x1;
+    kRinse = 0x2;
+  }
+
+  readonly attribute optional OCTET_STRING spinSpeeds[] = 0;
```
Comment (andy31415) [inline src/controller/data_model/controller-clusters.matter:2953]: spin speeds ios `string` in the spec, not octet_string (which would be binary data).

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    <name>Washer Controls</name>
+    <code>0x0053</code>
+    <define>WASHER_CONTROLS_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>This cluster supports remotely monitoring and controling the different typs of functionality available to a washing device, such as a washing machine.</description>
+
+    <globalAttribute side="server" code="0xFFFD" value="1" />
+
+    <attribute side="server" code="0x0000" define="SPIN_SPEEDS"               type="ARRAY" entryType="OCTET_STRING"                                writable="false" isNullable="false" optional="true">SpinSpeeds</attribute>
```
Comment (marktrayer) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:37]: Shouldn't OCTET_STRING be CHAR_STRING?

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    <attribute side="server" code="0x0001" define="SPIN_SPEED_CURRENT"        type="INT8U"                                min="0x00" max="0x1F"    writable="true"  isNullable="true"  optional="true">SpinSpeedCurrent</attribute>
+    <attribute side="server" code="0x0002" define="NUMBER_OF_RINSES"          type="INT8U"                                                         writable="true"  isNullable="true"  optional="true">NumberOfRinses</attribute>
+    <attribute side="server" code="0x0003" define="MAXRINSES"                 type="INT8U"                                                         writable="false"                    optional="true">MaxRinses</attribute>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:40]: not sure if we want to have min and max here. Spec is restrictive here (0 to 8)

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <bitmap name="WasherControlFeature" type="BITMAP32">
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:20]: Name this `Feature` based on us trying to standardize naming. We ended up asking everyone to just use `Feature` (we had some back and forth here)

**Review by andy31415 — APPROVED** [review]

**Pull request merged**